### PR TITLE
Modify dependencies info to include the full dependency path instead of just the version

### PR DESCRIPTION
**Pull request by mpeddada1**
Modify the main build.gradle and related build files to reference the full dependency data instead of just the version. This change attempts to make our dependency info more friendly with dependabot, which is currently unable to parse through our dependencies.
For #2602

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -48,44 +48,46 @@ subprojects {
...
+  project.ext.dependencyStrings = [
     // For Google libraries, check <http-client-bom.version>, <google.auth.version>, <guava.version>,
     // ... in https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/pom.xml
     // for best compatibility.
-    GOOGLE_HTTP_CLIENT: '1.34.0',
-    GOOGLE_HTTP_CLIENT_APACHE_V2: '1.34.0',
-    GOOGLE_AUTH_LIBRARY_OAUTH2_HTTP: '0.18.0',
-    GUAVA: '28.1-jre',
-    JSR305: '3.0.2', // transitively pulled in by GUAVA
+    GOOGLE_HTTP_CLIENT: "com.google.http-client:google-http-client:1.34.0",
```
Comment (chanseokoh) [inline build.gradle:55]: Really super-nit, but we try to maintain consistency regarding `'` and `"`. In Gradle (which is Groovy language), we use `'` for static strings like this one, while using `"` for strings containing expressions to evaluate (such as `"some text with a ${variable}"`).

File: build.gradle [hunk 2]
```diff
@@ -48,44 +48,46 @@ subprojects {
...
+    JBCRYPT: "org.mindrot:jbcrypt:0.4",
   ]
 
   // Use this to ensure we correctly override transitive dependencies
   // TODO: There might be a plugin that does this
   task ensureTransitiveDependencyOverrides {
     def rules = [
-        'google-http-client': dependencyVersions.GOOGLE_HTTP_CLIENT,
-        'google-http-client-apache-v2': dependencyVersions.GOOGLE_HTTP_CLIENT_APACHE_V2,
+        'google-http-client': dependencyStrings.GOOGLE_HTTP_CLIENT.split(':')[-1],
```
Comment (loosebazooka) [inline build.gradle:89]: I wonder if this name `google-https-client` should be derived from the `GOOGLE_HTTP_CLIENT` value??

Maybe you could use a closure to create a map from a list?
rules = [GOOGLE_HTTP_CLIENT, GOOGLE_HTTP_CLIENT_APACHE_V2] .collectEntries {
  [calculate name: calculate version]
}
```

I'm only vaguely familiar with groovy transforms, might be worth looking up

File: jib-maven-plugin/build.gradle [hunk 3]
```diff
@@ -12,10 +12,10 @@ dependencies {
...
-  implementation "com.google.cloud.tools:jib-maven-plugin-extension-api:${dependencyVersions.MAVEN_EXTENSION}"
+  implementation dependencyStrings.MAVEN_EXTENSION
 
-  implementation "org.apache.maven:maven-plugin-api:${dependencyVersions.MAVEN_API}"
-  implementation "org.apache.maven:maven-core:${dependencyVersions.MAVEN_API}"
+  implementation dependencyStrings.MAVEN_API
+  implementation dependencyStrings.MAVEN_CORE
 
   // compileOnly + testImplementation equivalent to "provided"
   compileOnly 'org.apache.maven.plugin-tools:maven-plugin-annotations:3.6.0'
```
Comment (loosebazooka) [inline jib-maven-plugin/build.gradle:21]: It might be worth moving all these values into the root build.gradle?

File: build.gradle (hunk 2 above)
Comment (mpeddada1) [reply build.gradle]: Oh I didn't know you could do that, thanks!

File: jib-maven-plugin/build.gradle (hunk 3 above)
Comment (mpeddada1) [reply jib-maven-plugin/build.gradle]: Done

File: build.gradle [hunk 4]
```diff
@@ -52,43 +52,50 @@ subprojects {
...
+    JACKSON_DATAFORMAT_YAML: 'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.11.0',
+    ASM: 'org.ow2.asm:asm:7.3.1',
+    PICOCLI: 'info.picocli:picocli:4.2.0',
+
+    MAVEN_API: 'org.apache.maven:maven-plugin-api:3.5.2',
+    MAVEN_CORE: 'org.apache.maven:maven-core:3.5.2',
+    MAVEN_COMPAT: 'org.apache.maven:maven-compat:3.5.2',
+    MAVEN_ANNOTATIONS_LATEST: 'org.apache.maven.plugin-tools:maven-plugin-annotations:3.6.0',
+    MAVEN_ANNOTATIONS_OLD: 'org.apache.maven.plugin-tools:maven-plugin-annotations:3.5',
+    MAVEN_TESTING_HARNESS: 'org.apache.maven.plugin-testing:maven-plugin-testing-harness:3.3.0',
```
Comment (loosebazooka) [inline build.gradle:78]: can we move 78-79 to the test commented block below?

File: build.gradle [hunk 5]
```diff
@@ -52,43 +52,50 @@ subprojects {
...
 
   // Use this to ensure we correctly override transitive dependencies
   // TODO: There might be a plugin that does this
   task ensureTransitiveDependencyOverrides {
-    def rules = [
-        'google-http-client': dependencyStrings.GOOGLE_HTTP_CLIENT.split(':')[-1],
-        'google-http-client-apache-v2': dependencyStrings.GOOGLE_HTTP_CLIENT_APACHE_V2.split(':')[-1],
-    ]
+    def dependenciesList = [dependencyStrings.GOOGLE_HTTP_CLIENT, dependencyStrings.GOOGLE_HTTP_CLIENT_APACHE_V2]
+    def rules = dependenciesList.collectEntries{[/*name: */ it.split(':')[1], /*version: */ it.split(':')[-1]]}
```
Comment (loosebazooka) [inline build.gradle:98]: I think using `[2]` instead of `[-1]` will help us catch bugs better?

File: build.gradle (hunk 1 above)
Comment (mpeddada1) [reply build.gradle]: Thanks for pointing this out! Fixed the style issue.

File: jib-maven-plugin/build.gradle [hunk 6]
```diff
@@ -12,32 +12,32 @@ dependencies {
...
+  testImplementation dependencyStrings.SLF4J_API_OLD
+  testImplementation dependencyStrings.SYSTEM_RULES
 
-  testImplementation 'org.apache.maven.shared:maven-verifier:1.7.2'
-  testImplementation "org.apache.maven:maven-compat:${dependencyVersions.MAVEN_API}"
-  testImplementation 'org.slf4j:slf4j-api:1.7.30'
-  testImplementation 'org.slf4j:slf4j-simple:1.7.30'
+  testImplementation dependencyStrings.MAVEN_VERIFIER
+  testImplementation dependencyStrings.MAVEN_COMPAT
+  testImplementation dependencyStrings.SLF4J_API_LATEST
```
Comment (loosebazooka) [inline jib-maven-plugin/build.gradle:37]: hrmm weird that these are both in here, can you file a new issue to handle standardizing the SLF4J apis?

Comment (mpeddada1) [reply jib-maven-plugin/build.gradle:37]: Sure! Filed issue #2608

File: build.gradle (hunk 5 above)
Comment (mpeddada1) [reply build.gradle]: Good point, done.

File: build.gradle (hunk 4 above)
Comment (mpeddada1) [reply build.gradle]: Done.

File: build.gradle [hunk 7]
```diff
@@ -48,45 +48,52 @@ subprojects {
...
 
   // Use this to ensure we correctly override transitive dependencies
   // TODO: There might be a plugin that does this
   task ensureTransitiveDependencyOverrides {
-    def rules = [
-        'google-http-client': dependencyVersions.GOOGLE_HTTP_CLIENT,
-        'google-http-client-apache-v2': dependencyVersions.GOOGLE_HTTP_CLIENT_APACHE_V2,
-    ]
+    def dependenciesList = [dependencyStrings.GOOGLE_HTTP_CLIENT, dependencyStrings.GOOGLE_HTTP_CLIENT_APACHE_V2]
+    def rules = dependenciesList.collectEntries{[/*name: */ it.split(':')[1], /*version: */ it.split(':')[2]]}
```
Comment (loosebazooka) [inline build.gradle:96]: Sorry this seems like python style comments, which is fine.
I think `/*name*/` , `/*version*/` is enough?

Comment (mpeddada1) [reply build.gradle]: Oh sorry, I removed the colon from the comment. Let me know if you want the comment to be moved outside of `collectEntries`

**Comment by loosebazooka** [comment]
Not sure why the travis builds aren't coming through? I'll re-trigger them

**Pull request merged**